src/foam/core/Action.js

```diff
@@ -177,10 +190,21 @@ If empty than no permissions are required.`,
     function maybeCall(x, data) {
       var self = this;
       function call() {
-        self.code.call(data, x, self);
+        var running = self.runningMap.get(data);
+        if ( ! running ) {
+          running = foam.core.SimpleSlot.create({ value: false });
+          if ( data ) self.runningMap.set(data, running);
+        }
+        if ( running.get() ) { console.warn("Attempted to call action that is in progress."); }
```

Comment:
This should probably be a call to x.warn().

Should we return early here and prevent calling the action?  Bit of a design decision to make here about the failure case.

On the one hand if we don't call the action that's in progress we protect against data binding and other code mistakes that might trigger the action twice in quick succession.  On the other hand, if the action returns a promise that never resolves, then the user will be unable to try the action again until, possibly ever.

Think about which failure case we care more about and which provides the better user experience, then add a comment here describing why we still call the action or why not.

Reply:
I think the problem of quick successive action trigger can easily be handled by a debounce which I think ActionView.js is already doing. Having User stuck forever on the other hand is less desirable.